### PR TITLE
Ding ming he branch

### PR DESCRIPTION
**Pull request by domhe**
This `Pull Request` is to manage Dingming He's branch

### CODE REVIEW

**Comment by MrYuan123** [comment]
Please Try: https://docs.scala-lang.org/getting-started/sbt-track/testing-scala-with-sbt-on-the-command-line.html

File: 7200_Final_TechIndex/src/test/scala/ReadDataSpec.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    appleData(0)._1.getClass.getSimpleName shouldBe("LocalDate")
+    appleData(0)._2.getClass.getSimpleName shouldBe("double")
+
+  }
+
+
+  behavior of "trimToRegion"
+  it should "get the data in the time we want" in {
+    val readData = new ReadData()
+    val appleFilename = "data/stocks/AAPL.csv"
```
Comment (MrYuan123) [inline 7200_Final_TechIndex/src/test/scala/ReadDataSpec.scala:21]: You don't need to use all of the data to test, just use one specific file to test the function

File: 7200_Final_TechIndex/src/test/scala/ReadDataSpec.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    val appleFile  = new File(appleFilename)
+    val appleData = readData.readFile(appleFile)
+
+    val formatter = DateTimeFormatter.ofPattern("d-MMM-yy");
+    val dateStrN1 = "1-Jan-18";
```
Comment (MrYuan123) [inline 7200_Final_TechIndex/src/test/scala/ReadDataSpec.scala]: You can parse some special date so we can test the edge condition
